lib/handler.js

```diff
@@ -43,6 +47,20 @@ const handler = async (req, res, pgPool) => {
       res,
       pgPool,
       fetchRetrievalSuccessRate)
+  } else if (req.method === 'GET' && segs[0] === 'participants' && segs[1] === 'daily' && segs.length === 2) {
+    await getStatsWithFilterAndCaching(
+      pathname,
+      searchParams,
+      res,
+      pgPool,
+      fetchDailyParticipants)
+  } else if (req.method === 'GET' && segs[0] === 'participants' && segs[1] === 'monthly' && segs.length === 2) {
```

Comment:
```suggestion
  } else if (req.method === 'GET' && pathname === '/participants/monthly') {
```

no need to use segments if we have static urls

Reply:
(also applies above and below)

Reply:
Yeah, I was wondering about this too.

But then your suggestion changes how we handle paths like `/participants//monthly` and thus it would introduce inconsistency with the rest of our API 😢 

Maybe I can have a "normalised path string" created via `segments.join('/')` and us that.

WDYT?

Reply:
If it gets complicated like this, let's just switch to fastify 🤷‍♂️  I don't mind all these edge cases, since I don't yet see any problems arise from them. If you want this to be tighter, I propose let's just make the switch

Reply:
TBH, I don't have an appetite to spend more time on this. I know the current code could be improved; maybe we can also simplify it by not worrying about inconsistencies in edge case handling. Either way, I feel it's a minor detail, and we have bigger fish to fry.

I am going to land what we have now; I added the "Migrate our REST APIs to Fastify" task to our [M4.2 milestone](https://github.com/filecoin-station/roadmap/issues/61).